golang/Duration.dhall

```diff
@@ -1,15 +1,25 @@
-let Duration = < Seconds : Natural | Minutes : Natural | Hours : Natural >
+let Duration =
+      let Unit = < Seconds | Minutes | Hours >
+
+      in  { Type = { unit : Unit, magnitude : Natural }
+          , Unit
+          , Seconds =
+              λ(magnitude : Natural) → { magnitude, unit = Unit.Seconds }
+          , Minutes =
+              λ(magnitude : Natural) → { magnitude, unit = Unit.Minutes }
+          , Hours = λ(magnitude : Natural) → { magnitude, unit = Unit.Hours }
+          }
 
 let render =
       let render
-          : ∀(value : Duration) → Text
-          =   λ(value : Duration)
-            → merge
-                { Seconds = λ(it : Natural) → "${Natural/show it}s"
-                , Minutes = λ(it : Natural) → "${Natural/show it}m"
-                , Hours = λ(it : Natural) → "${Natural/show it}h"
+          : ∀(value : Duration.Type) → Text
+          = λ(value : Duration.Type) →
+              merge
+                { Seconds = "${Natural/show value.magnitude}s"
+                , Minutes = "${Natural/show value.magnitude}m"
+                , Hours = "${Natural/show value.magnitude}h"
                 }
-                value
+                value.unit
```

Comment:
I want to remove the use of `merge` function as it affects:
1) Complexity
2) Performace

It should be changed to:

```
let Duration =
      let Unit = < Seconds | Minutes | Hours >

      in  { Type = { unit : Unit, magnitude : Natural }
          , Unit
          , Seconds =
              λ(magnitude : Natural) → { result = "${Natural/show magnitude}s", unit = Unit.Seconds }
```
Then you should run:

```

dhall <<< "let p = ./golang/package.dhall in p.Duration.Seconds 15"

{ result = "15s", unit = < Hours | Minutes | Seconds >.Seconds }
```

Reply:
I disagree - 
1. why even bother to keep the `Unit.Seconds` then?
2. if the whole point is just to get a `Text` representation, why do we need to import this to begin with? Why not just write `Text` at the source?
3. We don't always want the rendered "15s" value, for example if we're writing a Prometheus Rule then we will need access to the `15 : Natural`. If we start straight with the `Text` representation then we design ourselves into a corner.
4. As this is a very small union it does not meaningfully affect performance and to assume otherwise is premature optimization. See also efforts to alias all declared types: https://github.com/dhall-lang/dhall-haskell/issues/1890#issuecomment-654547593 in which case there will not be a performance impact

Reply:
So
1) To be honest I don't see any real reason anymore
2) We don't, we can treat it as Text
3) Currently render always return 15s for seconds, so this point is irrelevant as you either need to create new type or create new function, both cases puts ourselves into the same corner
4) I don't want to have union at all, and when the times come we will replace all of them. Once you give yourself a slack in one place you'll do it every place/